go/vt/vtgate/planbuilder/ordering.go

```diff
@@ -289,10 +289,35 @@ func planRouteOrdering(orderBy sqlparser.OrderBy, node *route) (logicalPlan, err
 		if colNumber == -1 {
 			return nil, fmt.Errorf("unsupported: in scatter query: order by must reference a column in the select list: %s", sqlparser.String(order))
 		}
+		starColFixedIndex := colNumber
+		if selectStatement, ok := node.Select.(*sqlparser.Select); ok {
+			for i, selectExpr := range selectStatement.SelectExprs {
+				if starExpr, ok := selectExpr.(*sqlparser.StarExpr); ok {
+					if i < colNumber {
+						tableName := starExpr.TableName
+						tableMap := node.resultColumns[i].column.st.tables
+						var tableMeta *table
+						if tableName.IsEmpty() && len(tableMap) == 1 {
+							for j := range tableMap {
+								tableMeta = tableMap[j]
+							}
+						} else {
+							tableMeta = tableMap[tableName]
+						}
+						if tableMeta == nil {
+							break
+						}
+						starColFixedIndex += len(tableMeta.columnNames) - 1
```

Comment:
Why `+=` here? How would this work for a query that has multiple star expressions, like:

```sql
select t.*, r.*, t.col from t,r order by t.col
```

Reply:
Ah, I understand now, yeah, this would work even for multiple star expressions...